### PR TITLE
[Merged by Bors] - chore: add a `Repr` for `Sym2`

### PR DESCRIPTION
**Pull request by eric-wieser**
This matches the approach used by `Multiset`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Sym/Sym2.lean [hunk 1]
```diff
@@ -764,4 +764,8 @@ instance [IsEmpty α] : IsEmpty (Sym2 α) :=
 instance [Nontrivial α] : Nontrivial (Sym2 α) :=
   diag_injective.nontrivial
 
+-- TODO: use a sort order if available, gh-18166
```
Comment (semorrison) [inline Mathlib/Data/Sym/Sym2.lean:767]: Let's use full URLs when referencing issues or prs.

Actually, what is this, the number is too high to be an issue??

Comment (eric-wieser) [reply Mathlib/Data/Sym/Sym2.lean:767]:
```suggestion
-- TODO: use a sort order if available, https://github.com/leanprover-community/mathlib/issues/18166
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**